fission/src/ui/modals/mirabuf/ImportMirabufModal.tsx

```diff
@@ -3,80 +3,142 @@ import Modal, { ModalPropsImpl } from "@/components/Modal"
 import { FaPlus } from "react-icons/fa6"
 import Button from "@/components/Button"
 import Label, { LabelSize } from "@/components/Label"
-import { Data, Folder, Hub, Item, Project, getFolderData, getHubs, getProjects } from "@/aps/APSDataManagement"
+import { Data, Folder, Hub, Project, getHubs, getProjects, searchRootForMira } from "@/aps/APSDataManagement"
+import MirabufCachingService, { MirabufCacheInfo, MiraType } from "@/mirabuf/MirabufLoader"
+import APS from "@/aps/APS"
 
 interface ItemCardProps {
     id: string
     name: string
     buttonText: string
+    secondaryButtonText?: string
     onClick: () => void
+    secondaryOnClick?: () => void
 }
 
-const ItemCard: React.FC<ItemCardProps> = ({ id, name, buttonText, onClick }) => {
+const ItemCard: React.FC<ItemCardProps> = ({ id, name, buttonText, secondaryButtonText, onClick, secondaryOnClick }) => {
     return (
         <div
             key={id}
             className="flex flex-row align-middle justify-between items-center bg-background rounded-sm p-2 gap-2"
         >
             <Label className="text-wrap break-all">{name}</Label>
             <Button value={buttonText} onClick={onClick} />
```

Comment:
Could you make the buttons icons instead of text and make them the size of the icons, instead of spanning across the div.